### DISCUSSION TITLE
Access to params / searchParams / URL in app dir template

### DISCUSSION THREAD

**Discussion by mormahr** [vercel/next.js#46991] [Ideas]
### Goals

1. Adding metadata (e.g. OpenGraph images), without editing all individual pages
2. Adding navigation components, that need server access (e.g. loading next/previous pages from the database)

### Non-Goals

1. Changing the behavior of layouts

### Background

This came up with regard to layouts, discussed in #43704.
My primary use case is a site with lots of mdx pages (100+), where most of the metadata could be generated automatically. Another use case is previous / next navigation, requiring individual URLs pulled from the server.
As far as I know there's no way to add elements to the `<head>` with app directory pages. Otherwise it would have been possible to use a layout with a client component, to at least add metadata that doesn't rely on server data (e.g. OpenGraph image URLs).

Unfortunately it seems like metadata in templates doesn't work at all, at the moment. `generateMetadata` is called, but the results don't end up in the HTML. #46864

### Proposal

Make `params` and `searchParams` available to template components as well as `generateMetadata` in template files. Since templates are, by definition, server-rendered for each page, this should be possible.

**Comment by jiri-grussmann** [comment]
Over  a year and still no answer or fix... :(

**Reply by maiconsanson** [reply]
Another thing is it seems that the template is not rerender on navigation too.
https://github.com/vercel/next.js/issues/52422#issuecomment-2132258784
